onnxruntime/core/providers/shared_library/provider_interfaces.h

```diff
@@ -0,0 +1,476 @@
+  virtual Provider_KernelDefBuilder& SinceVersion(int since_version) = 0;
+  virtual Provider_KernelDefBuilder& Provider(const char* provider_type) = 0;
+  virtual Provider_KernelDefBuilder& TypeConstraint(const char* arg_name, MLDataType supported_type) = 0;
+
+  virtual std::unique_ptr<Provider_KernelDef> Build() = 0;
+
+  void operator=(const Provider_KernelDefBuilder& v) = delete;
+};
+
+using NodeIndex = size_t;
+using Provider_NodeAttributes = std::unordered_map<std::string, ONNX_NAMESPACE::Provider_AttributeProto_Copyable>;
+
+using Provider_InitializedTensorSet = std::unordered_map<std::string, const ONNX_NAMESPACE::Provider_TensorProto*>;
+
+struct Provider_NodeArg {
+  virtual ~Provider_NodeArg() = default;
+  virtual const std::string& Name() const noexcept = 0;
+  virtual const ONNX_NAMESPACE::Provider_TensorShapeProto* Shape() const = 0;
+  virtual ONNX_NAMESPACE::DataType Type() const noexcept = 0;
+
+  void operator=(const Provider_NodeArg& v) = delete;
+};
+
+struct Provider_Node {
+  virtual ~Provider_Node() = default;
+
+  virtual const std::string& OpType() const noexcept = 0;
+
+  virtual ConstPointerContainer<std::vector<Provider_NodeArg*>> InputDefs() const noexcept = 0;
+  virtual ConstPointerContainer<std::vector<Provider_NodeArg*>> OutputDefs() const noexcept = 0;
+  virtual NodeIndex Index() const noexcept = 0;
+
+  virtual const Provider_NodeAttributes& GetAttributes() const noexcept = 0;
+  virtual size_t GetInputEdgesCount() const noexcept = 0;
+  virtual size_t GetOutputEdgesCount() const noexcept = 0;
+
+  struct Provider_NodeIterator {
+    virtual ~Provider_NodeIterator() {}
+    virtual bool operator!=(const Provider_NodeIterator& p) const = 0;
+
+    virtual void operator++() = 0;
+    virtual const Provider_Node& operator*() = 0;
+  };
+
+  struct NodeConstIterator {
+    NodeConstIterator(std::unique_ptr<Provider_NodeIterator> p) : impl_{std::move(p)} {}
+
+    bool operator==(const NodeConstIterator& p_other) const;
+    bool operator!=(const NodeConstIterator& p_other) const {
+      return *impl_ != *p_other.impl_;
+    }
+
+    void operator++() {
+      impl_->operator++();
+    }
+    void operator--();
+
+    const Provider_Node& operator*() const {
+      return impl_->operator*();
+    }
+    const Provider_Node* operator->() const;
+
+    std::unique_ptr<Provider_NodeIterator> impl_;
+  };
+
+  NodeConstIterator InputNodesBegin() const noexcept { return NodeConstIterator(InputNodesBegin_internal()); }
+  NodeConstIterator InputNodesEnd() const noexcept { return NodeConstIterator(InputNodesEnd_internal()); }
+
+  virtual std::unique_ptr<Provider_NodeIterator> InputNodesBegin_internal() const noexcept = 0;
+  virtual std::unique_ptr<Provider_NodeIterator> InputNodesEnd_internal() const noexcept = 0;
+};
+
+#ifndef PROVIDER_BRIDGE_ORT
+// if we are export the fused function to dll, the function will still in the same binary as lotus
```

Comment:
probably should remove the reference to 'lotus' in the comment

Reply:
Ah, it's from the execution_provider.h comment. I removed this comment and will fix that one too.